src/util/getRoutePath.js

```diff
@@ -1,10 +1,14 @@
-export default function getRoutePath(route, routes) {
+export default function getRoutePath(route, routes, name) {
   const matchIndex = routes.indexOf(route);
   if (matchIndex < 0) {
     throw new Error('`route` not found in `routes`');
   }
   const routesUntil = routes.slice(0, matchIndex + 1);
-  return routesUntil
+  let routePath = routesUntil
     .reduce((acc, { path }) => acc.concat(path), [])
     .join('/');
+  if (name) {
+    routePath = `${routePath}@${name}`;
```

Comment:
Maybe one of the unwise ones?

> Other characters are excluded because gateways and other transport
   agents are known to sometimes modify such characters, or they are
   used as delimiters.
   unwise      = "{" | "}" | "|" | "\" | "^" | "[" | "]" | "`"

http://www.ietf.org/rfc/rfc1738.txt
http://www.ietf.org/rfc/rfc2396.txt

A good candidate seems to be pipe.

```js
routePath = `${routePath}|||${name}`;
```